src/services/projectService.ts

```diff
@@ -293,8 +327,18 @@ export default class ProjectService implements IProjectService {
         updatedProject.tags = existingTags;
     }
 
-    private async addMissingTags(project: IProject, tags: ITag[]) {
-        const missingTags = tags.filter((fileTag) => !project.tags.find((tag) => fileTag.name === tag.name ));
+    private async addMissingTagsAndUpdateLabelCount(project: IProject, tags: ITag[], tagLabelCount?: any) {
+        const missingTags = tags.filter((fileTag) => {
```

Comment:
use map

Reply:
Actually, I don't think using map is optimal in this case because we only want to return the missing tags